templates/_components/pagination.twig

```diff
@@ -0,0 +1,143 @@
+{# @var object pageInfo - Pass the pageInfo object from Craft's {% paginate %} tag to pre-fill all props #}
+{% set pageInfo = pageInfo ?? null %}
+
+{#
+   @var int maxLinks
+   Controls the number of page links to show. Includes the first and last page URLs.
+   Odd numbers work best.
+
+   maxLinks has no effect when using manual props
+#}
+{% set maxLinks = maxLinks ?? 5 %}
+
+{# @var string|null nextUrl #}
+{% set nextUrl = nextUrl ?? pageInfo.nextUrl ?? null %}
+
+{# @var string|null prevUrl #}
+{% set prevUrl = prevUrl ?? pageInfo.prevUrl ?? null %}
+
+{# @var int currentPage #}
+{% set currentPage = currentPage ?? pageInfo.currentPage ?? null %}
+
+{# @var int totalPages #}
+{% set totalPages = totalPages ?? pageInfo.totalPages ?? null %}
+
+{# @var string|null firstUrl #}
+{% set firstUrl = firstUrl ?? pageInfo.firstUrl ?? null %}
+
+{# @var string|null lastUrl #}
+{% set lastUrl = lastUrl ?? pageInfo.lastUrl ?? null %}
+
+{# @var array visibleUrls An array of visible page URLs indexed by page number #}
+{% set visibleUrls = visibleUrls ?? [] %}
+
+{# @var string size #}
+{% set size = size ?? 'md' %}
+
+{# @var string ariaLabel #}
+{% set ariaLabel = ariaLabel ?? 'pagination navigation' %}
+
+{# Computed values #}
+{% set needEllipsis = totalPages > maxLinks %}
+{% set elipsisThreshold = floor(maxLinks / 2) %}
+{% set hasStartEllipsis = needEllipsis and currentPage > maxLinks - elipsisThreshold %}
+{% set hasEndEllipsis = needEllipsis and currentPage < totalPages - elipsisThreshold %}
+
+{% if pageInfo %}
+  {% if hasStartEllipsis %}
+    {% set maxLinks = maxLinks - 1 %}
+  {% endif  %}
+
+  {% if hasEndEllipsis or (currentPage < totalPages - (elipsisThreshold - 1)) %}
+    {% set maxLinks = maxLinks - 1 %}
+  {% endif %}
+
+  {% set visibleUrls = pageInfo.getDynamicRangeUrls(maxLinks) %}
+{% endif %}
+
+{% macro paginationItem(props = {}) %}
+  {% set url = props.url ?? null %}
+  {% set page = props.page ?? null %}
+  {% set size = props.size ?? null %}
+  {% set currentPage = props.currentPage ?? null %}
+
+  {{ include('_components/button.twig', {
+      href: url,
+      text: page,
+      variant: page == currentPage ? 'contained' : 'subtle',
+      size: size,
+      class: cx('!p-0 aspect-square rounded-full'),
```

Comment:
The Blueprint component used some size classes on the buttons to make them square for pagination. 

I didn't love that the height of the button and these size classes were spread apart in two separate files. Felt like it would be easy to tweak heights in buttons.js and forget to update the pagination. 

https://github.com/vigetlabs/blueprint/blob/main/src/components/navigation/Pagination.astro#L64-L66

By using `aspect-square` here it was possible to make the buttons square while letting the heights from buttons.js define the overall size. 

@andrewmosby It's super minor... but wasn't sure if you'd like to fold that back in to blueprint.